File: python/problem050.py
import math

number_array = [True] * 1000000

for i in range(2, int(math.sqrt(1000000))):
    if number_array[i]:
        j = i + i
        while j < 1000000:
            number_array[j] = False
            j += i

prime_numbers = []

for i in range(2, 1000000):
    if number_array[i]:
        prime_numbers.append(i)

max_count = 0
result = 0
for a in prime_numbers:
    value = 0
    count = 0
    for b in range(prime_numbers.index(a), len(prime_numbers)):
        count += 1
        value += prime_numbers[b]
        if value > 1000000:
            break
        if prime_numbers.count(value) != 0:
            if max_count < count:
                max_count = count
                result = value

    check = 0
    for x in range(prime_numbers.index(a), prime_numbers.index(a) + max_count):
        check += prime_numbers[x]
    if check > 1000000:
        break

print(result)

File: javascript/problem050.js
var array = new Array(1000000);

for(i=2; i<Math.sqrt(100000); i++){
  if(array[i] == undefined){
    for(j=i+i; j<1000000; j = j + i){
      array[j] = false;
    }
  }
}

var primeNumbers = [];
for(i=2; i<1000000; i++){
  if(array[i] == undefined){
    primeNumbers.push(i);
  }
}

var maxCount = 0;
var result = 0;
for(a in primeNumbers){
  var value = 0;
  var count = 0;
  for(b = a; b < primeNumbers.length; b++){
    count = count + 1;
    value = value + primeNumbers[b];
    if(value > 1000000){
      break;
    }
    if(primeNumbers.indexOf(value) != -1){
      if(maxCount < count){
        maxCount = count;
        result = value;
      }
    }
  }

  var temp = parseInt(a);
  var check = 0;
  for(x=a; x<temp+maxCount; x++){
    check += primeNumbers[x]
  }
  if(check > 1000000){
    break;
  }
}

console.log(result);
